### ISSUE TITLE
DL3022 false positives, where alias is defined in same Dockerfile

### ISSUE THREAD

**Issue by inickles-grapl** [hadolint/hadolint#761]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

After updating to version 2.8.0 we started getting unexpected DL3022 warnings. These appear to be false positives, as the alias exists within the same Dockerfile.

Example: 
The `COPY --from=build` here:
- https://github.com/grapl-security/grapl/blob/6d2bf19643d1fe07aca1b8cd3e9c978a049d0ef3/src/rust/Dockerfile#L190
References `FROM base AS build` here:
- https://github.com/grapl-security/grapl/blob/6d2bf19643d1fe07aca1b8cd3e9c978a049d0ef3/src/rust/Dockerfile#L43

### Actual behavior

I'm getting: 
```
DL3022 warning: COPY --from should reference a previously defined FROM alias
```

### Steps to reproduce the behavior

```bash
$ wget https://raw.githubusercontent.com/grapl-security/grapl/6d2bf19643d1fe07aca1b8cd3e9c978a049d0ef3/src/rust/Dockerfile
$ sed -e 's/hadolint ignore=DL3022//g' -i Dockerfile
$ docker run --rm -i hadolint/hadolint:v2.8.0 < Dockerfile
-:190 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:199 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:208 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:217 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:226 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:235 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:244 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:253 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:262 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:271 DL3022 warning: COPY --from should reference a previously defined FROM alias
-:281 DL3022 warning: COPY --from should reference a previously defined FROM alias
$ sed -n '43,43p' Dockerfile
FROM base AS build
$ sed -n '190,190p' Dockerfile
COPY --from=build /outputs/analyzer-dispatcher /
$ docker run --rm -i hadolint/hadolint:v2.8.0 hadolint --version
Haskell Dockerfile Linter v2.8.0-0-g398770f-dirty
```
